packages/ifpack2/src/Ifpack2_BlockTriDiContainer_impl.hpp

```diff
@@ -1065,30 +1069,15 @@ namespace Ifpack2 {
             const local_ordinal_type iend = pack_td_ptr(packptr(k+1));
             const local_ordinal_type diff = iend - ibeg;
             const local_ordinal_type icount = diff/3 + (diff%3 > 0);
-	    Kokkos::parallel_for(Kokkos::TeamThreadRange(member,icount),[&](const local_ordinal_type &ii) {
-		Kokkos::parallel_for(Kokkos::ThreadVectorRange(member, vector_length_value),[&](const int &v) {
+            Kokkos::parallel_for(Kokkos::ThreadVectorRange(member, vector_loop_size),[&](const int &v) {
+                Kokkos::parallel_for(Kokkos::TeamThreadRange(member,icount),[&](const local_ordinal_type &ii) {
                     const local_ordinal_type i = ibeg + ii*3;
                     for (local_ordinal_type j=0;j<blocksize;++j) 
                       values(i,j,j,v) = 1;
```

Comment:
Do we build this with complex Scalar types?  If so, I don't think this assignment (from an integer `1` to `impl_scalar_type`, which could be `Kokkos::complex<double>` or `Kokkos::complex<float>`) will compile.  The easy fix is to change the number to `1.0`.

Reply:
This is okay even for complex. Although nobody uses this with complex but I tested with complex before submitting PR.